<a href="https://colab.research.google.com/github/ohjisu320/study_data_analytics/blob/main/docs/EDA/EDA_LetalCarOfContractType.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## EDA 각 과정에서 insight 도출
- 제안 6: 고객 성향에 따른 제품 추천 시스템 개발
    - 문제 정의: 고객의 마일리지, 계약 유형, 나이 등을 분석하여 개인별 맞춤형 제품을 추천합니다.
    - 배경: 고객의 선호도와 행동을 분석하여 맞춤형 제품을 추천함으로써, 고객 만족도를 높이고 매출 증대에 기여할 수 있습니다.


### 기술 데이터 분석(DDA)

In [3]:
import pandas as pd
df_LCOCT = pd.read_csv('LetalCarOfContractType.csv')
df_LCOCT.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51304 entries, 0 to 51303
Data columns (total 17 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 51304 non-null  int64  
 1   type_of_contract   51300 non-null  object 
 2   type_of_contract2  51303 non-null  object 
 3   channel            51304 non-null  object 
 4   datetime           51304 non-null  object 
 5   Term               51304 non-null  int64  
 6   payment_type       51304 non-null  object 
 7   product            51303 non-null  object 
 8   amount             51304 non-null  int64  
 9   state              51304 non-null  object 
 10  overdue_count      51304 non-null  int64  
 11  overdue            51302 non-null  object 
 12  credit rating      42521 non-null  float64
 13  bank               48544 non-null  object 
 14  cancellation       51279 non-null  object 
 15  age                40509 non-null  float64
 16  Mileage            405

In [5]:
df_LCOCT.dropna(inplace=True)

In [8]:
# 계약 유형(범주형) 분포 확인
df_LCOCT['type_of_contract'].value_counts()
# 렌탈이 멤버십보다 10배 이상 많음 >> 대부분 고객은 렌탈 유형을 선택함.
# 인사이트 : 렌탈 유형에 대한 고객 선호도가 높음을 반영한 -> 마케팅, 프로모션 전략을 고려할 필요 있음

렌탈     38445
멤버십     2035
Name: type_of_contract, dtype: int64

In [9]:
# 고객 연령(연속형) 분포 확인
df_LCOCT['age'].describe()
# 소비 고객 연령 대부분 중년층(46세 전후)에 모여 있음
# 인사이트 : 한정된 연령대 소비자가 있으므로, 소비자 층을 넓히기 위한 프로모션 전략 수립 필요

count    40480.000000
mean        46.610944
std         13.464950
min         20.000000
25%         37.000000
50%         46.000000
75%         56.000000
max        107.000000
Name: age, dtype: float64

In [11]:
# 마일리지 분포 확인
df_LCOCT['Mileage'].describe()
# 평균 마일리지는 약  1967.97이며, 대부분이 Q1과 3 이하며 최소 872이고 최대 3803 사이 분포
# 인사이트 : 고객 마일리지 사용 패턴을 분석해 로열티 프로그램을 보다 많이 개발 활용

count    40480.000000
mean      1967.974753
std        436.694248
min        872.000000
25%       1646.750000
50%       1946.000000
75%       2263.000000
max       3803.000000
Name: Mileage, dtype: float64

### EDA
- 추가 확인 및 인사이트 재조정

## 보고용 insight 작성



- 계약 유형 별 고객 분포 인사이트
    + 분석 결과: 대부분 고객은 렌탈 계약 유형을 선택
    + 전략 제안: '렌탈' 계약에 대한 추가 마케팅 활동 강화. 관련 고객에 대한 맞춤형 제안을 개발해 고객 만족도 상승 필요
- 마일리지 분포 따른 인사이트
    + 분석 결과 : 사용 패턴이 중앙값(1646.75~2263.00)에 분포해 있음
    + 전략 제안 : 고객 마일리지 사용 패턴을 분석해 로열티 프로그램을 개선